src/viser/_gui_api.py

```diff
@@ -498,6 +505,81 @@ def add_gui_markdown(
         handle.content = content
         return handle
 
+    def add_gui_plotly(
+        self,
+        figure: go.Figure,
+        aspect_ratio: float = 1.0,
+        order: Optional[float] = None,
+        visible: bool = True,
+        font: Optional[str] = None,
+    ) -> GuiPlotlyHandle:
+        """Add a Plotly Plot to the GUI.
+
+        Args:
+            figure: Plotly figure to display.
+            aspect_ratio: Aspect ratio of the plot in the control panel (width / height).
+            order: Optional ordering, smallest values will be displayed first.
+            visible: Whether the component is visible.
+            font: Optional font to use for the plot.
+
+        Returns:
+            A handle that can be used to interact with the GUI element.
+        """
+        handle = GuiPlotlyHandle(
+            _gui_api=self,
+            _id=_make_unique_id(),
+            _visible=visible,
+            _parent_container_id=self._get_container_id(),
+            _order=_apply_default_order(order),
+            _figure=figure,
+            _aspect_ratio=aspect_ratio,
+            _font=font,
+        )
+
+        # If plotly.min.js hasn't been sent to the client yet, the client won't be able
+        # to render the plot. Send this large file now! (~3MB)
+        if not self._setup_plotly_js:
+            # Check if plotly is installed.
+            try:
+                import plotly
+            except ImportError:
+                raise ImportError(
+                    "You must have the `plotly` package installed to use the Plotly GUI element."
+                )
+
+            # Check that plotly.min.js exists.
+            plotly_path = (
+                Path(plotly.__file__).parent / "package_data" / "plotly.min.js"
+            )
+            assert (
+                plotly_path.exists()
+            ), f"Could not find plotly.min.js at {plotly_path}."
+
+            # Send it over!
+            with open(plotly_path, "r") as f:
+                plotly_js = f.read()
```

Comment:
since we're using pathlib already we can just do `plotly_js.read_text()`

Reply:
Ok! Updated. https://github.com/nerfstudio-project/viser/blob/3af36b1bb606bbb26bd59737c4d44eb6333cecc7/src/viser/_gui_api.py#L559-L563